datadotworld/client/api.py

```diff
@@ -242,17 +257,52 @@ def replace_dataset(self, dataset_key, **kwargs):
         except _swagger.rest.ApiException as e:
             raise RestApiError(cause=e)
 
+    def delete_dataset(self, dataset_key):
+        """Deletes a dataset and all associated data
+
+        Parameters
+        ----------
+        dataset_key : str
+            Dataset identifier, in the form of owner/id
+
+        Returns
+        -------
+        str
+            Success message for deleted dataset
+
+        Raises
+        ------
+        RestApiException
+            If a server error occurs
+
+        Examples
+        --------
+        >>> import datadotworld as dw
+        >>> api_client = dw.api_client()
+        >>> api_client.delete_dataset(
+        ...     'jonloyens/an-intro-to-dataworld-dataset')
+        >>> del_dataset.message
+        'Dataset has been successfully deleted.'
+        """
+        owner_id, dataset_id = parse_dataset_key(dataset_key)
+        try:
+            return self._datasets_api.delete_dataset(owner_id, dataset_id)
+        except _swagger.rest.ApiException as e:
+            raise RestApiError(cause=e)
+
     # File Operations
 
-    def add_files_via_url(self, dataset_key, files={}):
+    def add_files_via_url(self, dataset_key, files={}, **kwargs):
         """Add or update dataset files linked to source URLs
 
         Parameters
         ----------
         dataset_key : str
             Dataset identifier, in the form of owner/id
         files : dict
```

Comment:
If I understand correctly, a single description and set of labels can be provided and will be applied to all files in `files`. 

If that's right, then 1) the documentation isn't very accurate and 2), most importantly, I don't think that is what we want. My suggestion would be to allow users to use one of two types of values in the `files` dict. A simple url or a dict with `url`, `description` and `labels`. 

Alternatively, instead of introducing a `**kwargs`, we could introduce a `file_metadata` param of type dict, where keys are file names and values are again dict, containing `description` and `label`.